### ISSUE TITLE
tests are failing with >=libfaketime-0.9.7

### ISSUE THREAD

**Issue by Whissi** [rsyslog/rsyslog#2099]
Tests are failing like
```
[now_family_utc]: test $NOW family of system properties
------------------------------------------------------------
Test: ./now_family_utc.sh
------------------------------------------------------------
find: ‘/usr/ajp_temp’: Permission denied
find: ‘/usr/portage/packages/dev-db’: Permission denied
find: ‘/usr/portage/packages/dev-util’: Permission denied
find: ‘/usr/portage/packages/dev-python’: Permission denied
find: ‘/usr/portage/packages/app-portage’: Permission denied
find: ‘/usr/portage/packages/net-misc’: Permission denied
find: ‘/usr/portage/packages/games-fps’: Permission denied
find: ‘/usr/portage/packages/app-arch’: Permission denied
find: ‘/usr/portage/packages/dev-perl’: Permission denied
Testing '/usr/lib64/libfaketime.so.1' library ...
Test passed! Will use '/usr/lib64/libfaketime.so.1' library!
Note: System is year-2038 compliant
ABORT! Timeout waiting on startup (pid file)
connect(): Connection refused
connect() failed
error in trying to open connection i=0
error opening connections
error during tcpflood! see rsyslog.out.log.save for what was written
cp: cannot stat 'rsyslog.out.log': No such file or directory
FAIL now_family_utc.sh (exit status: 1)
```
and

```
[now-utc]: test $NOW-UTC
ERROR: left-over previous instance 42186, killing it
UID         PID   PPID  C STIME TTY          TIME CMD
portage   42186      1  0 Nov23 pts/6    00:00:00 ../tools/rsyslogd -C -n -irsyslog.pid -M../runtime/.libs:../.libs -f./testsuites/now_family_utc.conf
------------------------------------------------------------
Test: ./now-utc.sh
------------------------------------------------------------
find: ‘/usr/ajp_temp’: Permission denied
find: ‘/usr/portage/packages/dev-db’: Permission denied
find: ‘/usr/portage/packages/dev-util’: Permission denied
find: ‘/usr/portage/packages/dev-python’: Permission denied
find: ‘/usr/portage/packages/app-portage’: Permission denied
find: ‘/usr/portage/packages/net-misc’: Permission denied
find: ‘/usr/portage/packages/games-fps’: Permission denied
find: ‘/usr/portage/packages/app-arch’: Permission denied
find: ‘/usr/portage/packages/dev-perl’: Permission denied
Testing '/usr/lib64/libfaketime.so.1' library ...
Test passed! Will use '/usr/lib64/libfaketime.so.1' library!
Note: System is year-2038 compliant
ABORT! Timeout waiting on startup (pid file)
connect(): Connection refused
connect() failed
error in trying to open connection i=0
error opening connections
error during tcpflood! see rsyslog.out.log.save for what was written
cp: cannot stat 'rsyslog.out.log': No such file or directory
FAIL now-utc.sh (exit status: 1)
```
[...]

I remember that I already saw these failures when we added tests using libfaketime to rsyslog for the first time when I tested libfaketime's git master at this time.

**Comment by rgerhards** [comment]
Thanks for the report. Given the fact that we have a very special use case for libfaketime, I wonder if we should cache an old version of libfaketime for our needs. TBH it looks like libfaketime drifts into a direction that makes it harder and harder for us to use. I am not sure if it is worth the trouble to fix this every time.

Thoughts?

**Comment by Whissi** [comment]
As distribution: No, please don't bundle anything. Even if it is just for testing.

I'll bisect libfaketime to understand what change is causing the problem. Maybe it is just a bug which will be fixed upstream. If we know the reason, we can re-evaluate the situation.

**Comment by rgerhards** [comment]
excellent, sounds like a very good plan! Looking forward to the news. As I understand, it is not yet urgent, right?

**Comment by Whissi** [comment]
Right, not urgent. Adoption of libfaketime-0.9.7 is low at the moment, https://repology.org/metapackage/libfaketime/versions (but keep in mind it was released this week).

**Comment by rgerhards** [comment]
good!

**Comment by Whissi** [comment]
```
$ git bisect log
git bisect start
# bad: [3ed13f498c4f2fda6f429a493c7877380aac6e46] Fix libfaketimeMT build to define PTHREAD
git bisect bad 3ed13f498c4f2fda6f429a493c7877380aac6e46
# good: [3bed636a412585cf542305cd6283c9cf9a66b9cf] Debian Bug#753460: Updated fix for 699599
git bisect good 3bed636a412585cf542305cd6283c9cf9a66b9cf
# good: [6c207c9c6886033980e78d48493bbf522da4b964] Custom timestamp filename added: ~/.faketimerc alternative
git bisect good 6c207c9c6886033980e78d48493bbf522da4b964
# bad: [0bde083556e243e87bddaaf94e68f2ef85dad769] Make sure that the constructor has been called when the loader chose to load another constructor first.
git bisect bad 0bde083556e243e87bddaaf94e68f2ef85dad769
# good: [10b479cf29d59092b20aaccf4fe8cb1aab4d9482] Merge pull request #59 from mac-joker/master
git bisect good 10b479cf29d59092b20aaccf4fe8cb1aab4d9482
# first bad commit: [0bde083556e243e87bddaaf94e68f2ef85dad769] Make sure that the constructor has been called when the loader chose to load another constructor first.

bde083556e243e87bddaaf94e68f2ef85dad769 is the first bad commit
commit 0bde083556e243e87bddaaf94e68f2ef85dad769
Author: Erich E. Hoover
Date:   Wed Dec 10 14:55:01 2014 -0700

    Make sure that the constructor has been called when the loader chose to load another constructor first.

:040000 040000 060a1da4a6378d732d111a1cc5a9c937b4b467b4 6e455c8cb8a9db3fb98de92e46cf804a49fba959 M
```

**Comment by Whissi** [comment]
Only happens when you build *rsyslog* with jemalloc (`--enable-jemalloc`), jemalloc-5.0.1 in my case.

**Comment by rgerhards** [comment]
@Whissi do you have a recommendation on how to proceed forward? Just curios...

**Comment by Whissi** [comment]
Confirmed upstream bug, there's not much we can do: You don't want to enforce <libfaketime-0.9.7 **in** *rsyslog* because once this is fixed in *libfaketime*, user/maintainer would have to patch the check in *rsyslog* to get *rsyslog* working with the patched version. As package manager you will restrict newer *libfaketime* versions. In Gentoo I am doing https://github.com/gentoo/gentoo/blob/master/app-admin/rsyslog/rsyslog-8.31.0.ebuild#L99 for example.

You maybe want to maintain a "known issue" log where you document such an issue.

**Comment by rgerhards** [comment]
> You maybe want to maintain a "known issue" log where you document such an issue.

sounds good, will do!

**Comment by jaydev3646** [comment]
Hi Team,

I am using RSYSLOG-8.24 and my Cisco Devices are not working. I can see it is listening on UDP and TC port 514 but not collecting any log. Please help me in this . I am not sure how the conf file should look like.

**Comment by rgerhards** [comment]
@jaydev3646 please open a new issue - what you posted has nothing to do with this issue here. I will delete it to avoid confusion. Thx.
